test/characteristic_test.dart

```diff
@@ -0,0 +1,356 @@
+import 'dart:async';
+import 'dart:developer';
+import 'dart:typed_data';
+
+import 'package:flutter_ble_lib/flutter_ble_lib.dart';
+import 'package:flutter_ble_lib/src/_managers_for_classes.dart';
+import 'package:mockito/mockito.dart';
+import 'package:test/test.dart';
+
+import 'mock/manager_mock.dart';
+import 'mock/mock_peripheral.dart';
+import 'test_util/characteristic_generator.dart';
+import 'test_util/descriptor_generator.dart';
+
+class ServiceMock extends Mock implements Service {}
+
+void main() {
+  Peripheral peripheral = PeripheralMock();
+  ManagerForCharacteristic managerForCharacteristic =
+      ManagerForCharacteristicMock();
+  CharacteristicGenerator characteristicGenerator =
+      CharacteristicGenerator(managerForCharacteristic);
+  DescriptorGenerator descriptorGenerator =
+      DescriptorGenerator(ManagerForDescriptorMock());
+
+  Characteristic characteristic =
+      characteristicGenerator.create(123, ServiceMock());
+
+  DescriptorWithValue createDescriptor(int seed) =>
+      descriptorGenerator.create(seed, characteristic);
+
+  tearDown(() {
+    [
+      peripheral,
+      managerForCharacteristic,
+    ].forEach(clearInteractions);
+  });
+
+  test("descriptors returns descriptors provided by manager", () async {
+    //given
+    when(managerForCharacteristic.descriptorsForCharacteristic(characteristic))
+        .thenAnswer((_) => Future.value([
+              createDescriptor(0),
+              createDescriptor(1),
+              createDescriptor(2),
+            ]));
+
+    //when
+    var descriptors = await characteristic.descriptors();
+
+    //then
+    expect(
+        descriptors,
+        equals([
+          createDescriptor(0),
+          createDescriptor(1),
+          createDescriptor(2),
+        ]));
+  });
+
+  test("read returns expected value", () async {
+    //given
+    when(managerForCharacteristic.readCharacteristicForIdentifier(
+            any, characteristic, "123"))
+        .thenAnswer((_) => Future.value(Uint8List.fromList([1, 2, 3, 4])));
+
+    //when
+    var value = await characteristic.read(transactionId: "123");
+
+    //then
+    expect(value, equals(Uint8List.fromList([1, 2, 3, 4])));
+  });
+
+  test(
+      "read invokes manager with expected params when transactionId is specified",
+      () {
+    //when
+    characteristic.read(transactionId: "123");
```

Comment:
```suggestion
    characteristic.read(transactionId: "a123");
```

Reply:
Is it only about not using integers as transaction ids as they should be provided in a word-like form?

Reply:
Integers are reserved for the library and the user should not use them. Since tests might be used as a reference point for using the library, they should be written in a way that can be copy-pasted by new users.